pkg/extension/sumologicextension/extension.go

```diff
@@ -148,11 +126,18 @@ func (se *SumologicExtension) Start(ctx context.Context, host component.Host) er
 
 	// Set the transport so that all requests from se.httpClient will contain
 	// the collector credentials.
-	rt, err := se.RoundTripper(se.httpClient.Transport)
+	se.httpClient.Transport, err = se.RoundTripper(se.httpClient.Transport)
 	if err != nil {
 		return fmt.Errorf("couldn't create HTTP client transport: %w", err)
 	}
-	se.httpClient.Transport = rt
+
+	if !registrationDone {
+		se.logger.Info("Checking if locally retrieved credentials are still valid...")
+		if err := se.validateCredenials(ctx, colCreds.Credentials); err != nil {
+			return fmt.Errorf("locally stored credentials are invalid: %w", err)
+		}
+		se.logger.Info("Local collector credentials all good, starting up the collector")
+	}
```

Comment:
Could we move it to `getCredentials` or it will too much logic in that function

Reply:
The problem with that is that we "check" the credentials by issuing a heartbeat which requires collector credentials injected into http calls which is done in lines 129:132 with `se.RoundTripper(...)` and this can only be done when we obtain the credentials. Hence validating the credentials can only be done with a configured http client and transport.

Reply:
ok